app/src/main/java/org/wikipedia/page/ToCHandler.java

```diff
@@ -446,6 +449,7 @@ private void scrollToListSectionByOffset(int yOffset) {
         if (itemToSelect != currentItemSelected) {
             adapter.setHighlightedSection(itemToSelect);
             currentItemSelected = itemToSelect;
+            bridge.execute(JavaScriptActionHandler.prepareToScrollTo(adapter.getItem(currentItemSelected).getAnchor(), "{ highlight: false }"));
```

Comment:
To clarify: is this actually necessary? This means that the "References" section will expand itself while the user is _in the middle_ of scrolling using the thumb-circle.  That sounds jarring -- is it really desirable behavior?

Reply:
So... the whole reason we are even trying to use 'prepareToScrollTo' in the toc at all, was to expand sections that we were about to scroll to... but  I do see the point in it being jarring if you are just scrolling by, and the sections are just expanding unnecessarily. 

The decision to restrict "scroll_to_anchor" handler on PageFragment was somewhat tied to this...  the handler would asynchronously scroll to sections when thumb-scrolling, which was glitchy.. so in order to 'only expand but not scroll' when 'scroll_to_anchor' requests came from toc, I had to restrict the handler...

I will remove this prepareToScrollTo and remove constraints on the 'scroll_to_anchor' handler and remove extra scrollToSection calls..